play-v29/src/main/scala/com/gu/googleauth/actions.scala

```diff
@@ -185,13 +187,13 @@ trait LoginSupport extends Logging {
   /**
     * Handle the OAuth2 callback, which logs the user in and redirects them appropriately.
     *
-    * Also ensures the user belongs to the (provided) required Google Groups.
+    * Also ensures the user is in the correct Google Groups, as defined by the given GroupCheckConfig
     */
-  def processOauth2Callback(requiredGoogleGroups: Set[String], groupChecker: GoogleGroupChecker)
+  def processOauth2Callback(groupCheckConfig: GroupCheckConfig, groupChecker: GoogleGroupChecker)
```

Comment:
Instead of changing the signature of this method, I would be inclined to add a new method with this new signature, and keep the old method but marked as deprecated. Should hopefully make upgrades easier for other teams.